envoy/datadog_checks/envoy/envoy.py

```diff
@@ -144,6 +145,8 @@ def included_metrics(self, metric):
 
     @AgentCheck.metadata_entrypoint
     def _collect_metadata(self):
+        if not self.collect_server_info:
+            return
```

Comment:
It seems conflicting if metadata is enabled but collect_server_info is disabled. Either we can log a debug message (like "Skipping server info collection because `collect_server_info` was disabled") or we can use this as a tracker to intuitively stop attempting to collect metadata between check runs if it's unreachable? Perhaps there is a specific message we're seeing?

Reply:
Do you mean the case where someone might have `enable_metadata_collection: true` in their agent config and `collect_server_info:false` for the envoy check? I didn't know about the agent config option, that is a good point. 

The idea is to bypass metadata collection due to known limitations on some Envoy deployments, avoid log spam if user knows that this is expected to fail (because it would be configured manually), while not needing to disable metadata collection all together (perhaps they want to collect it for other checks?). Displaying a debug statement sounds like a good idea.
 
There is no specific message, the endpoint is just unreachable.  So although providing an intuitive solution that would not require any further user configuration might be nice, it would be difficult to avoid false positives where we cannot reach the endpoint for some other reason besides the underlying known limitations.

Reply:
IMO if the endpoint is unreachable, then we can check the response code and log a message in the first occurrence, then skip it after. It will attempt on the first run (after restart) so customers don't need to actively maintain a config option, wdyt?

Reply:
I think this approach is fine but if a user's envoy temporarily fails then it will raise a false positive, no? We would need to see repeated failures, rather than going off of first failure. Even then, repeated failures are not indicators of not wanting to collect_server_info. 

If envoy is down, check fails, so does metadata, and thus it gets disabled. When envoy goes back up, check will still be running but not collecting metadata anymore, right?